advisor/backend.go

```diff
@@ -61,32 +64,24 @@ type CommandDB interface {
 // these closes the database again.
 func Create() (CommandDB, error) {
 	var err error
-	t := &writer{}
+	t := &writer{
+		fn: dirs.SnapCommandsDB + "." + strutil.MakeRandomString(12) + "~",
```

Comment:
Any chance we could use ioutil.TempFile()/osutil.AtomicFile() for this? Both return os.File (or derivative), the path should be available via `.Name()`.

Reply:
I can't create  a bolt database from a file.

I could land a change on snapd/bolt to add that ability, at which point yes.